trigger.js

```diff
@@ -215,39 +224,76 @@ class TriggerHappy {
-            // test motion vs token diagonals
-            if (motion.intersectSegment([target.x, target.y, target.x + target.w, target.y + target.h])
-            || motion.intersectSegment([target.x, target.y + target.h, target.x + target.w, target.y])) {
-                update.x = target.center.x - tw;
-                update.y = target.center.y - th;
-                return true;
+            const tx = target.x;
+            const ty = target.y;
+            const tw = target.w;
+            const th = target.h;
+            const tgw = target.data.width; // target token width in grid units
+            const tgh = target.data.height; // target token height in grid units
+                // test motion vs token diagonals
+            if (tgw > 1 && tgh > 1 && tgw * tgh > 4) {
+                // big token so do boundary lines
+                var intersects = ( motion.intersectSegment([tx,      ty,      tx + tw, ty     ])
+                                || motion.intersectSegment([tx + tw, ty + th, tx + tw, ty     ])
+                                || motion.intersectSegment([tx + tw, ty + th, tx,      ty + th])
+                                || motion.intersectSegment([tx,      ty,      tx,      ty + th]))
+            } else  // just check the diagonals
+                var intersects = (motion.intersectSegment([tx,       ty,      tx + tw, ty + th])
+                               || motion.intersectSegment([tx,       ty + th, tx + tw, ty     ]));
+            if (intersects) {
+                if (tgw === 1 && tgh === 1) { // simple case size 1 target, return straight away.
+                    update.x = target.center.x - tkw;
+                    update.y = target.center.y - tkh;
+                    return true;
+                }
+                
+                let corners = [];
+                const gridSize = canvas.grid.size;
+                // Create a grid of the squares covered by the target token
+                [...Array(tgw).keys()].forEach(xc => [...Array(tgh).keys()].forEach(yc => 
+                    corners.push({x: target.data.x + xc * gridSize, y: target.data.y + yc * gridSize})
+                ))
+
+                // Find the closest square to the token start position that intersets the motion
+                const closest = corners.sort((a, b) => 
+                    Math.hypot(token.x + tkw - (a.x + gridSize / 2), token.y  + tkh - (a.y + gridSize / 2)) - Math.hypot(token.x + tkw - (b.x + gridSize / 2), token.y  + tkh - (b.y + gridSize / 2)));
+                for (let corner of closest) {
+                    if (motion.intersectSegment([corner.x, corner.y, corner.x + gridSize, corner.y + gridSize])
+                     || motion.intersectSegment([corner.x, corner.y + gridSize, corner.x + gridSize, corner.y])) {
+                        update.x = corner.x;
+                        update.y = corner.y;;
+                        return true;
+                    }
+                };

```

Comment:
this is the complication that I feel is not necessary. I think if it's a "capture" trap, it should be ok to capture the token to its center. A pit would make them fall and slide to the center or something, that's acceptable in my opinion and I'd prefer not to have all this code in the handler.
My suggestion is to keep your changes with the intersection for > 2x2 doing the 4 intersections, but then always capture the token to the center. If someone wants captures in smaller areas, they can use smaller tokens next to each other.

Reply:
Let's agree to disagree on the experience of trapping tokens - since I can understand and accept your point of view. But it does raise the question of whether the increased complexity of doing the box outline check is worth it. Since tokens will move into the trap in the edge cases we discussed and the move trigger will fire anyway.

Reply:
And just in case there is any confusion - I am completely happy with your decisions on this - there is no angst/irritation of any sort. Love  the module, enjoying the collaboration - may there be much more.